### ISSUE TITLE
Can't connect to Azure IoT hub while on 4G (WiFi works) using Microsoft.Azure.Devices nuget in Xamarin iOS Preview 12.16.1.24

### ISSUE THREAD

**Issue by michaeldimoudis** [xamarin/xamarin-macios#6997]
### Steps to Reproduce

1. Create an Azure IoT hub with a device, add credentials and build app here https://github.com/michaeldimoudis/raspberrypi-xamarin-iot/tree/master/app
2. Upload app to iPhone with iOS 12.4.1 (I tried on iPhone X and iPhone 7)
3. Connect phone to WiFi and make sure connecting, querying device twin, etc works
4. Turn off WiFi and connect via 4G, and connection to Azure IoT hub is now not possible
5. I even tried specifying `Amqp_WebSocket_Only` in `ServiceClient.CreateFromConnectionString()` https://github.com/michaeldimoudis/raspberrypi-xamarin-iot/blob/master/app/IotApp/Azure/AzureIoTHub.cs to no avail
6. This all works on 4G on current stable Xamarin iOS `12.14.0.114`

### Expected Behavior
To be able to connect to Azure IoT Hub via Microsoft.Azure.Devices nuget package on WiFi and 4G connections

### Actual Behavior
Can only connect to Azure IoT Hub via Microsoft.Azure.Devices nuget package on WiFi, not on 4G

### Environment

```
=== Visual Studio Community 2019 for Mac (Preview) ===

Version 8.3 Preview (8.3 build 1630)
Installation UUID: 220a6133-b6f0-40b9-a68c-579812886ca2
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.16.1.24 (d16-3 / 08809f5b)

	Package version: 604000190

=== Mono Framework MDK ===

Runtime:
	Mono 6.4.0.190 (2019-06/6b4b99e571b) (64-bit)
	Package version: 604000190

=== NuGet ===

Version: 5.3.0.6192

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.0.0-preview9-19423-09
	2.2.7
	2.2.5
	2.2.4
	2.2.0
	2.1.9
	2.1.5
	2.1.2
	2.1.1
	2.1.0
SDK: /usr/local/share/dotnet/sdk/3.0.100-preview9-014004/Sdks
SDK Versions:
	3.0.100-preview9-014004
	3.0.100-preview8-013656
	2.2.402
	2.2.203
	2.2.107
	2.2.106
	2.2.100
	2.1.505
	2.1.403
	2.1.400
	2.1.302
	2.1.301
	2.1.300
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/Current/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.11.16
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 10.3 (14492.2)
Build 10G8

=== Xamarin.Mac ===

Version: 5.16.1.24 (Visual Studio Community)
Hash: 08809f5b
Branch: d16-3
Build date: 2019-09-06 08:42:41-0400

=== Xamarin.iOS ===

Version: 12.16.1.24 (Visual Studio Community)
Hash: 08809f5b
Branch: d16-3
Build date: 2019-09-06 08:42:42-0400

=== Xamarin Designer ===

Version: 16.3.0.178
Hash: 725c87d57
Branch: remotes/origin/d16-3
Build date: 2019-08-29 22:57:50 UTC

=== Xamarin.Android ===

Version: 10.0.0.40 (Visual Studio Community)
Commit: xamarin-android/d16-3/080eaac
Android SDK: /Users/michael/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		6.0 (API level 23)
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.2
SDK Build Tools Version: 28.0.3

Build Information: 
Mono: mono/mono/2019-06@6434153d8b7
Java.Interop: xamarin/java.interop/d16-3@5836f58
LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/michael/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 1.4.0.64
Hash: f70cfd2
Branch: remotes/origin/d16-3
Build date: 2019-09-02 17:32:00 UTC

=== Android Device Manager ===

Version: 1.2.0.112
Hash: 791e68e
Branch: remotes/origin/d16-3
Build date: 2019-09-02 17:32:26 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 803001630
Git revision: 456143789d948850ef2518c7401bcd0637d88e99
Build date: 2019-09-11 21:24:24+00
Build branch: release-8.3-preview4
Xamarin extensions: 52c395755d16d1c7077d3568574a19af9a23b423

=== Operating System ===

Mac OS X 10.14.6
Darwin 18.7.0 Darwin Kernel Version 18.7.0
    Tue Aug 20 16:57:14 PDT 2019
    root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64


```

### Build Logs

n/a, app builds as expected

### Example Project (If Possible)

https://github.com/michaeldimoudis/raspberrypi-xamarin-iot/tree/master/app

**Comment by chamons** [comment]
Sorry for the delay in response. I believe this may have the same root cause as this [other issue](https://github.com/xamarin/xamarin-macios/issues/6762).

Can you install this [build](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/master/4a08c6c7b298e447f92159b919dc8ef50b8ed40d/858/package/xamarin.ios-12.15.0.321.pkg) and see if that fixes things?

**Comment by chamons** [comment]
Actually - Instead of that random master build you could also use the "Internal Preview" channel in VSfM to test the fix.

**Comment by spouliot** [comment]
The preview channel still (an update is coming) points to 12.16.1.24 (same version as you used) so it won't help right away.

The fix is also available in the `Xcode 11 Previews` channel (XI 13.0.x) if that's an option for you (i.e. did you need something in 12.16 preview?)

**Comment by michaeldimoudis** [comment]
Thanks for the response. I was just playing with the preview and noticed this issue, and was just concerned this bug will be in stable in soon.

I easily moved to stable branch to fix my app, but as long as Xamarin is aware and believe it won’t be an issue on the next stable telecast I’ll be at ease :)

When I get some time I’ll try XCode 11 preview or the other link and post my findings. 

Thank you
